### DISCUSSION TITLE
Several dependencies missed in udpates

### DISCUSSION THREAD

**Discussion by romakulakov** [renovatebot/renovate#10310] [Request Help] [answered]
Hi I have tried to configure a renovate [here](https://github.com/romakulakov/renovate_test/pull/1) and I have several questions:

1. Some dependencies are missed from all PRs, f.e. `@material-ui/core`, `@material-ui/icons`, `lodash`, `typescript` but I expect them in "[Client] Update all non-major dependencies"
2. Is it possible to extract all updates for specific libs in a separate PR, in my case I try to extract
```
{        
      "matchPackagePatterns": ["^axios"],        
      "matchUpdateTypes": ["major", "minor", "patch"],        
      "enabled": true,        
     "groupName": "axios"      
}
```
But  it still appears in "[Client] Update all non-major dependencies"(in a real project it would be private libs groping)
3. Is it possible to separate a patch release for a specific library?
I want to enable only path updates for `yup` but it doesn't work:
```
{
  "matchPackagePatterns": ["yup", "@types/yup"],
  "separateMinorPatch": true
},
{
  "matchPackagePatterns": ["yup", "@types/yup"],
  "matchUpdateTypes": ["major", "minor"],
  "enabled": false
}
```

Thx

**Comment by rarkins** [comment, marked as answer]
1. This "missed" PRs are probably due to a misunderstanding about the default `rangeStrategy` behavior. Please look it up in the docs.
2. Please clarify why you want this, because maybe there's a better way. Especially why you think putting major/minor/patch together is a good idea, because usually it isn't.
3. Try changing `matchPackagePatterns` to `matchPackageNames`

**Reply by romakulakov** [reply]
Thank you for your quick answer, yes you are right, this part of the docs totally explain it:

> For example, if your package.json specifies a value for left-pad of ^1.0.0 and the latest version on npmjs is 1.2.0, then Renovate won't change anything because 1.2.0 satisfies the range. If instead you'd prefer to be updated to ^1.2.0 in cases like this, then configure rangeStrategy to bump in your Renovate config.

The thing that confused me is why `Upgrade axios to ^0.21.0` and `Upgrade json-schema to ^0.3.0` were still present in PR? Is there some logic that still bump version if there is no major version for lib?

I upgraded my `renovate.json` with `rangeStrategy=bump`, and 1 and 3 questions were resolved.

But 2 is still present, you are right about major version, I shouldn't group it together, so I dropped it. But lets say, I want have 2 PRs for all major-patch version for all libs except `@material-ui` and a separate PR for `@material-ui` libs(use case f.e. I want to have PR for all libs once a week, and PR for a specific libs once a day)
I tried:
```
{
  "matchPackagePatterns": ["^@material-ui/"],
  "groupName": "material"
}
```
but now I have all major-minor versions under "[Client] Update material" and no separate PR.

**Reply by rarkins** [reply]
> The thing that confused me is why `Upgrade axios to ^0.21.0` and `Upgrade json-schema to ^0.3.0` were still present in PR? Is there some logic that still bump version if there is no major version for lib?

Because when the value is 0.x then `^0.21.0` means `>=0.21.0 <0.22.0` and not `>=0.21.0 <1.0.0`

> But lets say, I want have 2 PRs for all major-patch version for all libs except `@material-ui` and a separate PR for `@material-ui` libs(use case f.e. I want to have PR for all libs once a week, and PR for a specific libs once a day)

I don't understand this description, so would be unlikely to give a correct answer for it. Please try to explain it more carefully and/or detailed. For example I've never heard the term "major-patch version" be used before.

**Reply by romakulakov** [reply]
>Because when the value is 0.x then ^0.21.0 means >=0.21.0 <0.22.0 and not >=0.21.0 <1.0.0

 Thx for the explanation

>I don't understand this description, so would be unlikely to give a correct answer for it. Please try to explain it more carefully and/or detailed. For example I've never heard the term "major-patch version" be used before.

Sorry for messed explanation, I meant PR with minor(mistyped as major:)) and patch versions, anyway I was able to resolve my second issue with
```
{
  "matchPackagePatterns": ["^@material-ui/"],
  "groupSlug": "material",
  "groupName": "material"
}
```
So all my issues are resolved, thank you once more
